sum/front/src/components/contractInteraction.tsx

```diff
@@ -1,22 +1,48 @@
-import React, { ChangeEvent, useContext, useState } from "react";
-import { IAccount } from "@massalabs/wallet-provider";
-import { Args } from "@massalabs/massa-web3";
+import { ChangeEvent, useState, useEffect } from "react";
+import { Args, IClient, ClientFactory } from "@massalabs/massa-web3";
+import { IAccount, providers } from "@massalabs/wallet-provider";
 import Loader from "./Loader";
-import { MassaContext } from "../App";
-
-interface ContractInteractionProps {
-    account: IAccount;
-}
 
 const CONTRACT_ADDRESS =
     "AS12YrZxFisWZCKJpXLEYfSYzrSCS4bjoyKGeaviQMmb5zqfgXaML";
 
 export default function ContractInteraction() {
-    const { account } = useContext(MassaContext)!;
+    const [errorMessage, setErrorMessage] = useState<any>("");
+    const [client, setClient] = useState<IClient | null>(null);
+    const [account, setAccount] = useState<IAccount | null>(null);
+
     const [num1, setNum1] = useState<number>(0);
     const [num2, setNum2] = useState<number>(0);
     const [result, setResult] = useState<string>("");
     const [loading, setLoading] = useState<boolean>(false);
+    const [loadingGlobal, setLoadingGlobal] = useState(true);
+
+    useEffect(() => {
+        const registerAndSetProvider = async () => {
+          try {
+            let provider = (await providers(true, 10000))[0];
```

Comment:
I would prefer to use find instead of choosing the first element of the array

Reply:
Do you have any specific criteria in mind when suggesting the use of the 'find' function? Or are you recommending it simply to select the first account?

Reply:
> I would prefer to use find instead of choosing the first element of the array

Done